internal/action/run.go

```diff
@@ -0,0 +1,89 @@
+package action
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"reflect"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/sethvargo/go-githubactions"
+)
+
+// OutputString takes an interface and returns a stringified version
+func OutputString(i interface{}) (string, error) {
+	if i == nil {
+		return "", nil
+	}
+
+	t := reflect.TypeOf(i)
```

Comment:
Avoid reflection in high level code in favor of:

* Type assertions: https://go.dev/tour/methods/15
* Type switches: https://go.dev/tour/methods/16